pkg/writer.go

```diff
@@ -107,6 +142,108 @@ func (s *writerImpl) Close() (err error) {
 	return
 }
 
+type encodeResult struct {
+	buf   []byte
+	entry seekTableEntry
+}
+
+func (s *writerImpl) writeManyEncoder(ctx context.Context, ch chan<- encodeResult, frame []byte) func() error {
+	return func() error {
```

Comment:
I would assume that this function owns the channel at this point and needs to close it in all codepaths.  Maybe a defer?